Dockerfile

```diff
@@ -80,7 +79,7 @@ RUN set -ex \
     && (pip3 uninstall uritemplate.py --yes || true) \
     && pip3 install --no-cache-dir uritemplate.py==0.3.0 \
     # Fix: https://github.com/CenterForOpenScience/osf.io/pull/6783
-    && python3 -m compileall /usr/lib/python3.10 || true \
+    && python3 -m compileall /usr/lib/python3.12 \
```

Comment:
Not a dealbreaker for me, but I think the `|| true` was there to make it so the image can still build for local dev even if one of the requirements fails to install properly. An affordance to help troubleshoot / fix those exact sort of issues.  But my brain is getting fuzzy, I'll leave it up to y'all.

Reply:
Well, from what I see, we do not use this Dockerfile for development purposes, and it may  backfire us if this occurs in release build (compilation failed but we move on). Moreover, as I understand we have `requirements`  service in `docker-compose.yml` which is responsible for building dependencies for dev environment. Please, correct me if I'm mistaken

Reply:
* I agree with Oleh. We use the already build image locally
* I think we can remove comments in line 81 too, see
  * https://github.com/CenterForOpenScience/osf.io/commit/fc36671a52228b7afdfdb58c64ecaa492edab032
  * https://github.com/CenterForOpenScience/osf.io/pull/6783